src/v/cluster/data_migration_frontend.cc

```diff
@@ -303,6 +308,19 @@ frontend::get_migration(id migration_id) {
       });
 }
 
+ss::future<frontend::list_mountable_topics_result>
+frontend::list_mountable_topics() {
```

Comment:
does this need to hop over to `data_migrations_shard`? maybe not, but it's a common pattern for frontends.

Reply:
This one doesn't as it doesn't access any state. It belongs to frontend only for cohesion and to signal api boundaries to users.